bin/magpie-hive-daemon.sh

```diff
@@ -0,0 +1,123 @@
+
+# Third argument is start or stop
+
+if [ "$1X" == "X" ]
+then
+    echo "User must specify Hive configuration path"
+    exit 1
+fi
+
+if [ ! -d $1 ]
+then
+    echo "Hive configuration path is not a directory"
+fi
+
+if [ "$2X" == "X" ]
+then
+    echo "User must specify path to Hive as second argument"
+    exit 1
+fi
+
+if [ ! -d $2 ]
+then
+    echo "Hive path is not a directory"
+fi
+
+if [ "$3X" == "X" ]
+then
+    echo "User must specify start or stop as third argument"
+    exit 1
+fi
+
+if [ "$3" != "start" ] && [ "$3" != "stop" ]
+then
+    echo "User must specify start or stop as third argument"
+    exit 1
+fi
+
+# source the hive-env.sh file
+source $1/hive-env.sh
+myhostname=`hostname`
+
+if [ "$3" == "start" ]
+then
+ 
+  echo "Starting Hive Metastore on ${myhostname}"
+
+  $2/bin/hive --service metastore &> /dev/null &
+
+  sleep 30
+
+  export PATH=${PATH}:$2/bin
+  echo "Checking Hive Metastore settings"
+
+  # If the current node is different from the node where metastore is running the JDBC connector throws errors 
+  # need to make sure $HIVE_HOME/bin is in path, then call:
+  # $HIVE_HOME/bin/metatool -listFSRoot  <-- this will show where the metastore is now
+  # if that is different from the current node, run:
+  # $HIVE_HOME/bin/metatool -updateLocation <new location> <old location>  <-- this updates the metastore to the new node
+  # EX: $HIVE_HOME/bin/metatool -updateLocation hdfs://atec-n030:54310/usr/var/tmp/atc/ecosystem/shared/hive/metastore/hive_test_db.db hdfs://atec-n034:54310/usr/var/tmp/atc/ecosystem/shared/hive/metastore/hive_test_db.db
+
+  fsroot=`${2}/bin/metatool -listFSRoot | grep 'hdfs://'`
+  thishostname=`hostname`
+  for line in `echo $fsroot`; do
+    fshost=`echo "${line}"|cut -d / -f 3`
+    IFS=':' read -r -a tmparr <<<"$fshost"
+    name="${tmparr[0]}"
+    port="${tmparr[1]}"
+
+    if [[ "$name" != "$thishostname" ]]; then
+      echo "${thishostname} not found in ${fshost}, attempting to update"
+      newfshost="${line/$name/$thishostname}"
+      $2/bin/metatool -updateLocation ${newfshost} ${line}
+    else
+      echo "Metastore located at ${fshost}, update not required."
+    fi
+  done
+  
+  echo "Starting Hive server on ${myhostname}"
+  #$2/bin/hive --service hiveserver2 --hiveconf PORT=11112 &> /dev/null &
+  $2/bin/hive --service hiveserver2 &> /dev/null &
+
+else
+  echo "Stopping Hive Metastore on $myhostname"
+  # No nice way to do this, other than to find and kill the Hive and Metastore PIDs
+
+  # take the Metastore down first.  If Hive can't write to the Metastore, any process still active
+  # should fail, but won't cause harm to the DB.  
+  kill $(ps aux | grep '[o]rg.apache.hadoop.hive.metastore.HiveMetaStore' | awk '{print $2}')
+
+  echo "Stopping Hive server on $myhostname"
+  # attempt to kill the Hive by grep-ing for hiveconf...that string is unique to the process
+  kill $(ps aux | grep '[H]iveServer2' | awk '{print $2}')
+fi
```

Comment:
For consistency, could the same code style pattern that was in ```magpie-storm-daemon.sh``` be used.  Save off the pid of the daemon in a file.  Then the pid is available in that file later on for killing.